docs/rfcs/0007-synchronized-releases.md

```diff
@@ -0,0 +1,105 @@
+---
+layout: page
+title: Synchronizing the branching and versioning of Android Components with the Mozilla release trains
+permalink: /rfc/0007-synchronized-releases
+---
+
+* Start date: 2021-03-22
+* RFC PR: [#9973](https://github.com/mozilla-mobile/android-components/pull/9973)
+
+## Summary
+
+Synchronizing the branching and versioning of Android Components with the Mozilla release trains (GeckoView) to simplify release processes and integration by:
+
+* Having just one component implementing `concept-engine` with GeckoView
+* Having the `master` branch track only GeckoView Nightly
+* Creating branches and releases tracking matching GeckoView releases (e.g. GeckoView 89.0 -> Android Components 89.0)
+
+## Motivation
+
+Initially we introduced three `concept-engine` implementations for GeckoView, one for each GeckoView release channel:
+
+* `browser-engine-gecko-nightly`
+* `browser-engine-gecko-beta`
+* `browser-engine-gecko`
+
+ This allowed us to release Android Components independently from GeckoView and consumers could use the latest AC release regardless of which GeckoView channel they wanted to use it with.
+
+This flexibility comes at a cost and makes our release process and maintenance more difficult. For every new GeckoView release across all channels (nightly, beta, release) we want to ship a new release of Fenix (and with that Android Components). So there is already an implicit synchronization of releases, but it is manual and confusing. By making this dependency explicit through synchronized releases we can simplify our processes.
+
+## Explanation
+
+### One GeckoView component
+
+Instead of having three components using GeckoView (`browser-engine-gecko-nightly`, `browser-engine-gecko-beta`, `browser-engine-gecko`), we will have just one (`browser-engine-gecko`) and on `master` it will track the latest GeckoView Nightly from `mozilla-central`.
```

Comment:
> `master`

Might this also be a good time to think about moving from `master` to `trunk` or similar for the default branch?

Reply:
Yes, absolutely. Writing this document reminded me of that too. Unfortunately this is so entangled with our release engineering code, that it wasn't super straight-forward to change so far. :(
Fenix tracking issue: https://github.com/mozilla-mobile/fenix/issues/11427

Reply:
I think we should do this, but maybe it is a good idea to do a followup. This is already a rather big change. Switching the branch name for A-C may need another RFC just for that?